tests/garage/torch/modules/test_multi_headed_mlp_module.py

```diff
@@ -70,72 +37,129 @@
 ]
 
 
-def helper_make_inits(val):
+def _helper_make_inits(val):
+    """Return the function that initialize variable with val.
+
+    Args:
+        val (int): Value to initialize variable.
+
+    Returns:
+        lambda: Lambda function that initialize variable with val.
+
+    """
     return lambda x: nn.init.constant_(x, val)
 
 
 @pytest.mark.parametrize(
-    'input_dim, output_dim, hidden_sizes, output_w_init_vals', plain_settings)
+    'input_dim, output_dim, hidden_sizes, output_w_init_vals, n_heads',
+    plain_settings)
 def test_multi_headed_mlp_module(input_dim, output_dim, hidden_sizes,
-                                 output_w_init_vals):
-    module = MultiHeadedMLPModule(
-        n_heads=len(output_w_init_vals),
-        input_dim=input_dim,
-        output_dims=output_dim,
-        hidden_sizes=hidden_sizes,
-        hidden_nonlinearity=None,
-        hidden_w_init=nn.init.ones_,
-        output_nonlinearities=None,
-        output_w_inits=list(map(helper_make_inits, output_w_init_vals)))
-
-    input = torch.ones(input_dim)
-    outputs = module(input)
-
-    for i in range(len(outputs)):
+                                 output_w_init_vals, n_heads):
+    """Test Multi-headed MLPModule.
+
+    Args:
+        input_dim (int): Input dimension.
+        output_dim (int): Ouput dimension.
+        hidden_sizes (list[int]): Size of hidden layers.
+        output_w_init_vals (list[int]): Init values for output weights.
+        n_heads (int): Number of output layers.
+
+    """
+    module = MultiHeadedMLPModule(n_heads=n_heads,
+                                  input_dim=input_dim,
+                                  output_dims=output_dim,
+                                  hidden_sizes=hidden_sizes,
+                                  hidden_nonlinearity=None,
+                                  hidden_w_init=nn.init.ones_,
+                                  output_nonlinearities=None,
+                                  output_w_inits=list(
+                                      map(_helper_make_inits,
```

Comment:
isn't this equivalent to
```python
[lambda x: nn.init.constant_(x, val) for val in output_w_init_vals]
```
?

Reply:
Actually, the result would not be the same.
I already tried your suggestion but it produced unexpected result.
You can check [this](https://stackoverflow.com/questions/6076270/lambda-function-in-list-comprehensions) 

If I want to use the list comprehension instead of the original code, I need to write
[(lambda y: (lambda x: nn.init.constant_(x, val)))(val) for val in output_w_init_vals]